x-pack/plugins/observability_solution/investigate_app/server/clients/create_entities_es_client.ts

```diff
@@ -0,0 +1,144 @@
+  dataset: ENTITY_HISTORY,
+});
+export const CONTAINER_ENTITIES_LATEST_ALIAS = entitiesAliasPattern({
+  type: 'container',
+  dataset: ENTITY_LATEST,
+});
+export const CONTAINER_ENTITIES_HISTORY_ALIAS = entitiesAliasPattern({
+  type: 'container',
+  dataset: ENTITY_HISTORY,
+});
+type LatestAlias =
+  | typeof SERVICE_ENTITIES_LATEST_ALIAS
+  | typeof HOST_ENTITIES_LATEST_ALIAS
+  | typeof CONTAINER_ENTITIES_LATEST_ALIAS;
+
+type HistoryAlias =
+  | typeof SERVICE_ENTITIES_HISTORY_ALIAS
+  | typeof HOST_ENTITIES_HISTORY_ALIAS
+  | typeof CONTAINER_ENTITIES_HISTORY_ALIAS;
+
+export function cancelEsRequestOnAbort<T extends Promise<any>>(
+  promise: T,
+  request: KibanaRequest,
+  controller: AbortController
+): T {
+  const subscription = request.events.aborted$.subscribe(() => {
+    controller.abort();
+  });
+
+  return promise.finally(() => subscription.unsubscribe()) as T;
+}
+
+export interface EntitiesESClient {
+  search<TDocument = unknown, TSearchRequest extends ESSearchRequest = ESSearchRequest>(
+    indexName: string,
+    searchRequest: TSearchRequest
+  ): Promise<InferSearchResponseOf<TDocument, TSearchRequest>>;
+  msearch<TDocument = unknown, TSearchRequest extends ESSearchRequest = ESSearchRequest>(
+    allSearches: TSearchRequest[]
+  ): Promise<{ responses: Array<InferSearchResponseOf<TDocument, TSearchRequest>> }>;
+}
+
+export function createEntitiesESClient({
+  request,
+  esClient,
+}: {
+  request: KibanaRequest;
+  esClient: ElasticsearchClient;
+}) {
+  function search<TDocument = unknown, TSearchRequest extends ESSearchRequest = ESSearchRequest>(
+    indexName: string,
+    searchRequest: TSearchRequest
+  ): Promise<InferSearchResponseOf<TDocument, TSearchRequest>> {
+    const controller = new AbortController();
+
+    const promise = cancelEsRequestOnAbort(
+      esClient.search(
+        { ...searchRequest, index: [indexName], ignore_unavailable: true },
+        {
+          signal: controller.signal,
+          meta: true,
+        }
+      ) as unknown as Promise<{
+        body: InferSearchResponseOf<TDocument, TSearchRequest>;
+      }>,
+      request,
+      controller
+    );
+
+    return unwrapEsResponse(promise);
+  }
+
+  return {
+    async search<TDocument = unknown, TSearchRequest extends ESSearchRequest = ESSearchRequest>(
+      entityIndexAlias: LatestAlias | HistoryAlias,
+      searchRequest: TSearchRequest
+    ): Promise<InferSearchResponseOf<TDocument, TSearchRequest>> {
+      return search(entityIndexAlias, searchRequest);
+    },
+
+    async msearch<TDocument = unknown, TSearchRequest extends ESSearchRequest = ESSearchRequest>(
+      allSearches: TSearchRequest[]
+    ): Promise<{ responses: Array<InferSearchResponseOf<TDocument, TSearchRequest>> }> {
+      const searches = allSearches
+        .map((params) => {
+          const searchParams: [MsearchMultisearchHeader, MsearchMultisearchBody] = [
+            {
+              index: [SERVICE_ENTITIES_LATEST_ALIAS],
```

Comment:
This is copypasta. I'd like to remove the reference to the service alias in particular.